crates/oxc_linter/src/rules/jest/prefer_to_have_length.rs

```diff
@@ -0,0 +1,269 @@
+        mem_expr: &MemberExpression,
+        parsed_expect_call: &ParsedExpectFnCall<'a>,
+        kind: Option<&str>,
+        ctx: &LintContext<'a>,
+    ) -> String {
+        let mut formatter = ctx.codegen();
+
+        let Expression::Identifier(prop_ident) = mem_expr.object() else {
+            return formatter.into_code();
+        };
+
+        if let Expression::Identifier(ident) = &call_expr.callee {
+            formatter.print_str(ident.name.as_bytes());
+            formatter.print_str(b"(");
+            formatter.print_str(prop_ident.name.as_bytes());
+            formatter.print_str(b")");
+        }
+
+        if let Some(kind_val) = kind {
+            if kind_val == "ComputedMember" {
+                formatter.print_str(b"[\"not\"]");
+            } else if kind_val == "StaticMember" {
+                formatter.print_str(b".not");
+            }
+        }
+
+        formatter.print_str(b".toHaveLength(");
+
+        if parsed_expect_call.args.len() > 0 {
+            for arg in parsed_expect_call.args {
+                let Argument::Expression(Expression::NumericLiteral(num)) = arg else {
+                    continue;
+                };
+                formatter.print_str(num.raw.as_bytes());
+            }
+        }
+
+        formatter.print_str(b")");
+        formatter.into_code()
+    }
+}
+
+#[test]
+fn tests() {
+    use crate::tester::Tester;
+
+    let pass = vec![
+        ("expect.hasAssertions", None),
+        ("expect.hasAssertions()", None),
+        ("expect(files).toHaveLength(1);", None),
+        ("expect(files.name).toBe('file');", None),
+        ("expect(files[`name`]).toBe('file');", None),
+        ("expect(users[0]?.permissions?.length).toBe(1);", None),
+        ("expect(result).toBe(true);", None),
+        ("expect(user.getUserName(5)).resolves.toEqual('Paul')", None),
+        ("expect(user.getUserName(5)).rejects.toEqual('Paul')", None),
+        ("expect(a);", None),
+    ];
+
+    let fail = vec![
+        ("expect(files[\"length\"]).toBe(1);", None),
+        ("expect(files[\"length\"]).toBe(1,);", None),
+        ("expect(files[\"length\"])[\"not\"].toBe(1);", None),
+        ("expect(files[\"length\"])[\"toBe\"](1);", None),
+        ("expect(files[\"length\"]).not[\"toBe\"](1);", None),
+        ("expect(files[\"length\"])[\"not\"][\"toBe\"](1);", None),
+        ("expect(files.length).toBe(1);", None),
+        ("expect(files.length).toEqual(1);", None),
+        ("expect(files.length).toStrictEqual(1);", None),
+        ("expect(files.length).not.toStrictEqual(1);", None),
+    ];
+
+    let fix = vec![
+        ("expect(files[\"length\"]).not.toBe(1);", "expect(files).not.toHaveLength(1);", None),
+        // Todo: Currently, this code will remove the last comma in the `toHaveLength` call.
+        // (
+        //     "expect(files[\"length\"]).toBe(1,);",
+        //     "expect(files).toHaveLength(1,);",
+        //     None
+        // ),
```

Comment:
Maybe we can get the range of arguments' source text?

https://github.com/oxc-project/oxc/blob/a6717db4238735f305b623b9e1e8fa85797cd83c/crates/oxc_linter/src/rules/unicorn/no_instanceof_array.rs#L50

Reply:
Thanks, Done.